# Fire Stations Decision Support System


# Introduction

Fire cause many deaths and damage in cities. Although there are cutting edge  systems which detect fire at the beginning, if the fire is large scale, the response time of fire stations is very curicial for avoiding deaths and more damage. In order to minimize the response time we will investigate the location data and the coverage area of the fire stations of NYC.
  

Nowadays, lots of cities share their data by public, for example NYC share their data on data.cityofnewyork.us and there exact locations of the fire stations and equipments.

At the end of the project we are aim to identify places which has low fire station level. The local goverments and retailer companies will be interested in this project. Since they can take action for increasing coverage.

For this project we need a city which has more than one fire station and their geospatial locations.

# DATA

In this project we are going to use NYC fire stations data which was provided in https://data.cityofnewyork.us/widgets/hc8x-tcnd

I download data in to my computer as an csv file. Let's look into our data.

In [9]:
import pandas as pd 

df = pd.read_csv('FDNY_Firehouse_Listing.csv')
df

FacilityName  \
0                                   Engine 4/Ladder 15   
1                                  Engine 10/Ladder 10   
2                                             Engine 6   
3    Engine 7/Ladder 1/Battalion 1/Manhattan Boroug...   
4                                             Ladder 8   
5                                    Engine 9/Ladder 6   
6                      Engine 15/Ladder 18/Battalion 4   
7                                  Engine 28/Ladder 11   
8                                             Engine 5   
9                                            Engine 55   
10                                Ladder 20/Division 1   
11                      Engine 24/Ladder 5/Battalion 2   
12                                  Engine 33/Ladder 9   
13                                Ladder 3/Battalion 6   
14                                            Squad 18   
15                                 Engine 34/Ladder 21   
16                                           Engine 26   
17                      Engine 3/Ladder 12/Battalion 7   
18                                  Engine 1/Ladder 24   
19                                           Engine 14   
20                               Engine 16 / Ladder 7    
21                                           Engine 21   
22                      Engine 54/Ladder 4/Battalion 9   
23                                           Engine 23   
24                                            Rescue 1   
25                                 Engine 40/Ladder 35   
26              Ladder 25/District Office 4/Division 3   
27                                           Engine 74   
28                                           Engine 65   
29                   Engine 8 / Ladder 2 / Battalion 8   
..                                                 ...   
188                              Engine 302/Ladder 155   
189                 Engine 317/Ladder 165/Battalion 54   
190                              Engine 311/Ladder 158   
191                                         Engine 314   
192                              Engine 331/Ladder 173   
193                   Engine 264/Engine 328/Ladder 134   
194  Engine 265/Ladder 121/Battalion 47 EMS Station 47   
195                            Engine 266/Battalion 47   
196                              Engine 268/Ladder 137   
197                                         Engine 329   
198                                           Marine 9   
199                               Engine 155/Ladder 78   
200                             Ladder 79/Battalion 22   
201                                         Engine 156   
202                               Engine 163/Ladder 83   
203                               Engine 153/Ladder 77   
204                               Engine 165/Ladder 85   
205                               Engine 157/Ladder 80   
206                                         Engine 158   
207                               Engine 166/Ladder 86   
208  Engine 154/District Office 8/ Staten Island Bo...   
209                            Engine 152/Battalion 21   
210                               Engine 161/Ladder 81   
211                     Engine 160/Rescue 5/Division 8   
212                                         Engine 159   
213                  Engine 162/Ladder 82/Battalion 23   
214                               Engine 167/Ladder 87   
215                               Engine 164/Ladder 84   
216                          Engine 168/EMS Station 23   
217                               Engine 151/Ladder 76   

                    FacilityAddress        Borough  Postcode   Latitude  \
0                   42 South Street      Manhattan   10005.0  40.703466   
1                124 Liberty Street      Manhattan   10006.0  40.710072   
2                 49 Beekman Street      Manhattan   10038.0  40.710048   
3              100-104 Duane Street      Manhattan   10007.0  40.715463   
4             14 North Moore Street      Manhattan   10013.0  40.71975

In [10]:
df.shape

(218, 12)

In [11]:
df.columns

Index(['FacilityName', 'FacilityAddress', 'Borough', 'Postcode', 'Latitude',
       'Longitude', 'Community Board', 'Community Council ', 'Census Tract',
       'BIN', 'BBL', 'NTA'],
      dtype='object')

We remove the columns which we will not use in our analysis.

In [12]:
df=df.drop(['Community Board', 'Community Council ', 'Census Tract',
       'BIN', 'BBL', 'NTA'],axis=1)

Now we delete the empty rows.

In [20]:
df = df.dropna()

In [21]:
df.shape

(213, 6)

In [13]:
df.dtypes

FacilityName        object
FacilityAddress     object
Borough             object
Postcode           float64
Latitude           float64
Longitude          float64
dtype: object

In [14]:
df.groupby('Borough').head()

FacilityName  \
0                                   Engine 4/Ladder 15   
1                                  Engine 10/Ladder 10   
2                                             Engine 6   
3    Engine 7/Ladder 1/Battalion 1/Manhattan Boroug...   
4                                             Ladder 8   
48                    Engine 60/Ladder 17/Battalion 14   
49                                            Squad 41   
50                      Engine 71/Ladder 55/Division 6   
51                                 Engine 50/Ladder 19   
52                    Engine 92/Ladder 44/Battalion 17   
82   Engine 207/Ladder 110/Battalion 31/Division 11...   
83                                          Engine 226   
84                               Engine 205/Ladder 118   
85                                          Engine 224   
86                               Engine 279/Ladder 131   
148                              Engine 258/Ladder 115   
149                              Engine 325/Ladder 163   
150                 Engine 259/Ladder 128/Battalion 45   
151                                         Engine 260   
152                              Engine 261/Ladder 116   
198                                           Marine 9   
199                               Engine 155/Ladder 78   
200                             Ladder 79/Battalion 22   
201                                         Engine 156   
202                               Engine 163/Ladder 83   

                 FacilityAddress        Borough  Postcode   Latitude  \
0                42 South Street      Manhattan   10005.0  40.703466   
1             124 Liberty Street      Manhattan   10006.0  40.710072   
2              49 Beekman Street      Manhattan   10038.0  40.710048   
3           100-104 Duane Street      Manhattan   10007.0  40.715463   
4          14 North Moore Street      Manhattan   10013.0  40.719759   
48         341 East 143rd Street          Bronx   10454.0  40.813024   
49         330 East 150th Street          Bronx   10451.0  40.817502   
50            720 Melrose Avenue          Bronx   10455.0  40.820422   
51        1155 Washington Avenue          Bronx   10456.0  40.829652   
52            1259 Morris Avenue          Bronx   10456.0  40.834627   
82            172 Tillary Street       Brooklyn   11201.0  40.695983   
83              409 State Street       Brooklyn   11217.0  40.686982   
84             74 Middagh Street       Brooklyn   11201.0  40.700172   
85              274 Hicks Street       Brooklyn   11201.0  40.692948   
86           252 Lorraine Street       Brooklyn   11231.0  40.672335   
148            10-40 47th Avenue         Queens   11101.0  40.745286   
149            41-24 51st Street         Queens   11377.0  40.745335   
150      33-51 Greenpoint Avenue         Queens   11101.0  40.735961   
151            11-15 37th Avenue         Queens   11101.0  40.759696   
152          3720-22 29th Street         Queens       NaN        NaN   
198  Saint George Ferry Terminal  Staten Island   10301.0  40.644112   
199           14 Brighton Avenue  Staten Island   10301.0  40.637734   
200        1189 Castleton Avenue  Staten Island   10310.0  40.634150   
201                 412 Broadway  Staten Island   10310.0  40.631093   
202            875 Jewett Avenue  Staten Island   10314.0  40.615251   

     Longitude  
0   -74.007538  
1   -74.012523  
2   -74.005245  
3   -74.005938  
4   -74.006678  
48  -73.922395  
49  -73.920370  
50  -73.915778  
51  -73.907831  
52  -73.913310  
82  -73.983159  
83  -73.982876  
84  -73.992239  
85  -73.996971  
86  -73.999611  
148 -73.952150  
149 -73.913527  
150 -73.933706  
151 -73.940603  
152        NaN  
198 -74.072559  
199 -74.087746  
200 -74.122199  
201 -74.116574  
202 -74.130971

In [15]:
df.groupby('Borough').size()

Borough
Bronx            34
Brooklyn         66
Manhattan        48
Queens           50
Staten Island    20
dtype: int64

In [16]:
import folium

In order to visualize our code we use folium. First we draw circle to center of the new york.

In [18]:
longitude=-74.0153365
latitude=40.7149555
lmap = folium.Map(location=[latitude, longitude], zoom_start=12) 
folium.CircleMarker(
    [latitude, longitude],
    radius=12,
    color='red',
    popup='NYC',
    fill = True,
    fill_color = 'red',
    fill_opacity = 0.6
).add_to(lmap)

lmap


Now we are going to show the locations of the fire stations.

In [22]:
for lat, lng, in zip(df.Latitude, df.Longitude):
    lmap.add_child(
        folium.CircleMarker(
            [lat, lng],
            radius=10, # define how big you want the circle markers to be
            color='red',
            fill=True,
            fill_color='blue',
            fill_opacity=0.8
        )
    )
lmap

# Methodology


In this project we will direct our efforts on detecting areas of NYC that have low fire station density, particularly those with low number of fire stations.

In first step we have collected the required data: location and type (category) of every fire station in NYC  We have also identified fire stationss according to  https://data.cityofnewyork.us/widgets/hc8x-tcnd data.


In second step we classified our data according to the boroughs. 

In third step we analyze the descriptive statistics of the fire station numbers.

In [60]:
gb = df.groupby('Borough')    
dat1=list(gb)[0]#bronx


In [39]:
dat2=list(gb)[1]#brooklyn

In [48]:
dat3=list(gb)[2]#manhattan


In [50]:
dat4=list(gb)[3]#queens

In [52]:
dat5=list(gb)[4]#state Island


In [44]:
longitude=-74.0153365
latitude=40.7149555
cmap = folium.Map(location=[latitude, longitude], zoom_start=12) 

In [64]:
for lat, lng, in zip(dat1[1].Latitude, dat1[1].Longitude): #Bronx Fire Stations
    cmap.add_child(
        folium.CircleMarker(
            [lat, lng],
            radius=10,
            color='red',
            fill=True,
            fill_color='yellow',
            fill_opacity=0.6
        )
    )
    
cmap

In [66]:
for lat, lng, in zip(dat2[1].Latitude, dat2[1].Longitude): #brooklyn Fire Stations
    cmap.add_child(
        folium.CircleMarker(
            [lat, lng],
            radius=10,
            color='yellow',
            fill=True,
            fill_color='red',
            fill_opacity=0.6
        )
    )
    
cmap

In [67]:
for lat, lng, in zip(dat3[1].Latitude, dat3[1].Longitude): #manhattan Fire Stations
    cmap.add_child(
        folium.CircleMarker(
            [lat, lng],
            radius=10,
            color='green',
            fill=True,
            fill_color='blue',
            fill_opacity=0.6
        )
    )
    
cmap

In [68]:
for lat, lng, in zip(dat4[1].Latitude, dat4[1].Longitude): #queens Fire Stations
    cmap.add_child(
        folium.CircleMarker(
            [lat, lng],
            radius=10,
            color='blue',
            fill=True,
            fill_color='green',
            fill_opacity=0.6
        )
    )
    
cmap

In [69]:
for lat, lng, in zip(dat4[1].Latitude, dat4[1].Longitude): #state Island Fire Stations
    cmap.add_child(
        folium.CircleMarker(
            [lat, lng],
            radius=10,
            color='blue',
            fill=True,
            fill_color='red',
            fill_opacity=0.6
        )
    )
    
cmap

# Analysis

In [76]:
df.groupby('Borough').size().describe()

count     5.000000
mean     42.600000
std      16.516658
min      20.000000
25%      34.000000
50%      47.000000
75%      48.000000
max      64.000000
dtype: float64

According to number of fire stations mean is 42.6

Bronx,Staten Island has less fire station than mean that's why NYC can increase the number of fire stations in order to increase the coverage.

# Results and Conclusion 

In this report we make an decision support system in order to analyze the fire station coverage in New York City. There can be more detailed analysis but I have time shortage. According to our analysis there is no uniform distribution of fire stations numbers for each borough. 

Thank you for your review.